From ec96537bbf218e3cc0bd8a54c000b08f614df633 Mon Sep 17 00:00:00 2001
From: Shoji Keita <awaittrot@shjk.jp>
Date: Wed, 14 Jul 2021 07:17:18 +0900
Subject: arm64: dts: sun50i-a64-pinephone: Add mount-matrix for PinePhone
 magnetometers.

For lis3mdl, values are based on datasheet and PCB drawing.
For af8133j, values are from testing on a real device.

Signed-off-by: Shoji Keita <awaittrot@shjk.jp>
---
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
index 694d715ff875..80a08fa0a52a 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
@@ -506,6 +506,9 @@ af8133j: magnetometer@1c {
 		reset-gpios = <&pio 1 1 GPIO_ACTIVE_LOW>;
 		avdd-supply = <&reg_dldo1>;
 		dvdd-supply = <&reg_dldo1>;
+		mount-matrix = "0", "-1", "0",
+				"-1", "0", "0",
+				"0", "0", "-1";
 	};
 
 	/* Magnetometer */
@@ -514,6 +517,9 @@ lis3mdl: magnetometer@1e {
 		reg = <0x1e>;
 		vdd-supply = <&reg_dldo1>;
 		vddio-supply = <&reg_dldo1>;
+		mount-matrix = "0", "1", "0",
+				"-1", "0", "0",
+				"0", "0", "1";
 	};
 
 	/* Light/proximity sensor */
-- 
2.35.3

